src/PrestaShopBundle/Translation/Provider/ModulesProvider.php

```diff
@@ -23,43 +23,268 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
+declare(strict_types=1);
 
 namespace PrestaShopBundle\Translation\Provider;
 
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShop\PrestaShop\Core\Translation\Locale\Converter;
+use PrestaShopBundle\Translation\Loader\DatabaseTranslationLoader;
+use Symfony\Component\Translation\MessageCatalogue;
+
 /**
  * Translation provider for native modules (maintained by the core team)
  * Translations are provided by Crowdin.
  */
-class ModulesProvider extends AbstractProvider implements UseDefaultCatalogueInterface
+class ModulesProvider implements ProviderInterface
 {
+    public const DEFAULT_LOCALE = 'en-US';
+
+    /**
+     * @var DatabaseTranslationLoader
+     */
+    private $databaseLoader;
+
+    /**
+     * @var string the resource directory
+     */
+    protected $resourceDirectory;
+
+    /**
+     * @var string the Catalogue locale
+     */
+    protected $locale;
+
+    /**
+     * @var string the Catalogue domain
+     */
+    protected $domain;
+
+    public function __construct(DatabaseTranslationLoader $databaseLoader, $resourceDirectory)
+    {
+        $this->databaseLoader = $databaseLoader;
+        $this->resourceDirectory = $resourceDirectory;
+        $this->locale = self::DEFAULT_LOCALE;
+    }
+
     /**
      * {@inheritdoc}
      */
-    public function getTranslationDomains()
+    public function getDirectories(): array
     {
-        return ['^Modules[A-Z]'];
+        return [$this->getResourceDirectory()];
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getFilters()
+    public function getFilters(): array
     {
         return ['#^Modules[A-Z]#'];
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getIdentifier()
+    public function getTranslationDomains(): array
+    {
+        return ['^Modules[A-Z]'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLocale(): string
+    {
+        return $this->locale;
+    }
+
+    /**
+     * @param string $locale
+     */
+    public function setLocale(string $locale): self
+    {
+        $this->locale = $locale;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setDomain(string $domain): self
+    {
+        $this->domain = $domain;
+
+        return $this;
+    }
+
+    /**
+     * Get the PrestaShop locale from real locale.
+     *
+     * @return string The PrestaShop locale
+     *
+     * @deprecated since 1.7.6, to be removed in the next major
+     */
+    public function getPrestaShopLocale(): string
```

Comment:
Could you move these repetitive method to an abstract ?

Reply:
The goal is to get rib of abstract first to be able to refactor each class safely